std/security/ed-signer.go

```diff
@@ -0,0 +1,65 @@
+package security
+
+import (
+	"crypto/ed25519"
+	"time"
+
+	enc "github.com/named-data/ndnd/std/encoding"
+	basic_engine "github.com/named-data/ndnd/std/engine/basic"
+	"github.com/named-data/ndnd/std/ndn"
+	"github.com/named-data/ndnd/std/utils"
+)
+
+// edSigner is a signer that uses Ed25519 key to sign packets.
+type edSigner struct {
+	timer ndn.Timer
+	seq   uint64
+
+	keyLocatorName enc.Name
+	key            ed25519.PrivateKey
+	forCert        bool
+	forInt         bool
+	certExpireTime time.Duration
+}
+
+func (s *edSigner) SigInfo() (*ndn.SigConfig, error) {
```

Comment:
This function largely duplicates `rsaSigner.SigInfo` and `eccSigner.SigInfo`, except the assignment of `Type` field.
The logic shall be extracted to a common function like this:

```go
func (s *asymSigner) SigInfo(sigType ndn.SigType) (*ndn.SigConfig, error)
```

Fields needed by this function are moved into `asymSigner` struct, which is then embedded into `edSigner` struct.